### PR TITLE
Update Gramine Branch in config.yaml.template

### LINKED ISSUE #115: GSC `master` should use Gramine `master` branch, not `v1.3.1`
## Description
When gsc master is built for gramine v1.3.1 on dcap setup, it fails with error error: **Cannot open device /dev/sgx_enclave. Please make sure the Intel SGX kernel module is loaded.**

This failure is caused by the changes in  gsc repo for [SGX driver ](0b60fa28aab0c9bc1b907230879235870f600b66) and with the the config yaml template which was modified to build with in-kernel driver

For another description of the same problem, see https://github.com/gramineproject/gsc/issues/114.

## Steps to reproduce
1. Copy config.yaml.template to config.yaml
2. Specify Gramine commit as v1.3.1 in config.yaml
3. Follow the steps from [README](https://github.com/gramineproject/gsc/blob/master/test/README.rst)

## Expected results
**Since, we are building on dcap setup, the sgx mountings are /dev/sgx/enclave and /dev/sgx/provision., so the device path is updated in docker run command**

**docker run --device=/dev/sgx/enclave -v /var/run/aesmd/aesm.socket:/var/run/aesmd/aesm.socket gsc-ubuntu18.04-bash -c ls**

```
Gramine is starting. Parsing TOML manifest file, this may take some time...
-----------------------------------------------------------------------------------------------------------------------
Gramine detected the following insecure configurations:

  - loader.insecure__use_cmdline_argv = true   (forwarding command-line args from untrusted host to the app)

Gramine will continue application execution, but this configuration must not be used in production!
-----------------------------------------------------------------------------------------------------------------------
bin
boot
dev
etc
gramine
home
lib
lib64
media
mnt
opt
proc
root
run
sbin
srv
sys
tmp
usr
var
```

## Actual results
**docker run --device=/dev/sgx/enclave -v /var/run/aesmd/aesm.socket:/var/run/aesmd/aesm.socket gsc-ubuntu18.04-bash -c ls**
```
Gramine is starting. Parsing TOML manifest file, this may take some time...
error: Cannot open device /dev/sgx_enclave. Please make sure the Intel SGX kernel module is loaded.
error: load_enclave() failed with error -2
```

### PR DESCRIPTION
**Pull request by anjalirai-intel**
GSC Master is not compatible with Gramine v1.3.1. We must either use GSC v1.3.1 and Gramine v1.3.1, or both on master

Fixes #115

Signed-off-by: Anjali Rai <anjali.rai@intel.com>

## How to test this PR? 

- - -
This change is [Reviewable](https://reviewable.io/reviews/gramineproject/gsc/117)

### CODE REVIEW

**Review by dimakuv — CHANGES_REQUESTED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gsc/117)* status: all files reviewed, 3 unresolved discussions, not enough approvals from maintainers (3 more required), not enough approvals from different teams (2 more required, approved so far: ) (waiting on @anjalirai-intel)

___
*[`-- commits` line 2 at r1](https://reviewable.io/reviews/gramineproject/gsc/117#-NKma4s_ARUs0Snje2sB):*
I would change to:
```
Specify Gramine branch `master` in `config.yaml.template`

GSC master may currently fail if run with Gramine v1.3.1 and Linux 5.10 or less
(i.e. host Linux doesn't have in-kernel SGX driver). Thus, GSC master is not compatible
currently with v1.3.1, so we must specify `master` branch of Gramine.
```
___
*[`-- commits` line 7 at r1](https://reviewable.io/reviews/gramineproject/gsc/117#-NKmabBu2QO23cNa4irs):*
We never put these "fixes ..." lines in commit messages, only in PR descriptions. So this should be removed.
___
*[`config.yaml.template` line 18 at r1](https://reviewable.io/reviews/gramineproject/gsc/117#-NKmZyqn5bYByArdGScx:-NKmZyqn5bYByArdGScy:buk8wvw) ([raw file](https://github.com/gramineproject/gsc/blob/404d6ea32eb61bfe68022520ec983b17b64425f3/config.yaml.template#L18)):*
> ```
> #   Image:      "<prebuilt Gramine Docker image>"
> # In order to use specific tag with Gramine , we need to checkout similar tag on gsc for
> # compatability, if available. for e.g. Gramine v1.3.1 and GSC v1.3.1
> ```

This is not a hard requirement, it is a soft requirement. So it doesn't deserve to be written here, at least not in such a manner.

I would rephrase to this:
```
# 
# GSC releases are guaranteed to work with corresponding Gramine releases (and GSC master
# branch is guaranteed to work with current Gramine master branch). Therefore, in case of
# issues, specify Gramine Branch that corresponds to your current GSC release (e.g. "v1.3.1"
# for GSC v1.3.1).
```

(Please also rewrap my suggestion to 100-char-limit.)

**Comment by anjalirai-intel** [comment]
*[`-- commits` line 7 at r1](https://reviewable.io/reviews/gramineproject/gsc/117#-NKmabBu2QO23cNa4irs):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

We never put these "fixes ..." lines in commit messages, only in PR descriptions. So this should be removed.
</blockquote></details>

Done

**Comment by anjalirai-intel** [comment]
*[`-- commits` line 2 at r1](https://reviewable.io/reviews/gramineproject/gsc/117#-NKma4s_ARUs0Snje2sB):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

I would change to:
```
Specify Gramine branch `master` in `config.yaml.template`

GSC master may currently fail if run with Gramine v1.3.1 and Linux 5.10 or less
(i.e. host Linux doesn't have in-kernel SGX driver). Thus, GSC master is not compatible
currently with v1.3.1, so we must specify `master` branch of Gramine.
```
</blockquote></details>

Done.

**Comment by anjalirai-intel** [comment]
*[`config.yaml.template` line 18 at r1](https://reviewable.io/reviews/gramineproject/gsc/117#-NKmZyqn5bYByArdGScx:-NKqFDPUCwwmiFnfC81i:b-896fix) ([raw file](https://github.com/gramineproject/gsc/blob/404d6ea32eb61bfe68022520ec983b17b64425f3/config.yaml.template#L18)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

This is not a hard requirement, it is a soft requirement. So it doesn't deserve to be written here, at least not in such a manner.

I would rephrase to this:
```
# 
# GSC releases are guaranteed to work with corresponding Gramine releases (and GSC master
# branch is guaranteed to work with current Gramine master branch). Therefore, in case of
# issues, specify Gramine Branch that corresponds to your current GSC release (e.g. "v1.3.1"
# for GSC v1.3.1).
```

(Please also rewrap my suggestion to 100-char-limit.)
</blockquote></details>

Done.

**Review by dimakuv — CHANGES_REQUESTED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gsc/117)* status: all files reviewed, 1 unresolved discussion, not enough approvals from maintainers (3 more required), not enough approvals from different teams (2 more required, approved so far: ) (waiting on @anjalirai-intel)

___
*[`config.yaml.template` line 17 at r2](https://reviewable.io/reviews/gramineproject/gsc/117#-NKqd3rZ6CmiJXWmsnpZ:-NKqd3rZ6CmiJXWmsnp_:blnw2n9) ([raw file](https://github.com/gramineproject/gsc/blob/93818d5bc4041e8fabac9c48714e9c6e32c0c89f/config.yaml.template#L17)):*
> ```
> # the `gsc build-gramine` command). For this, remove Repository and Branch and instead write:
> #   Image:      "<prebuilt Gramine Docker image>"
> # GSC releases are guaranteed to work with corresponding Gramine releases (and GSC master
> ```

Please add an empty comment line before this `GSC releases ...` paragraph, for readability. Just like above, before the `It is also possible ...` paragraph.

**Comment by anjalirai-intel** [comment]
*[`config.yaml.template` line 17 at r2](https://reviewable.io/reviews/gramineproject/gsc/117#-NKqd3rZ6CmiJXWmsnpZ:-NKqhG8Y5gyIshuY2SAR:bb6un1j) ([raw file](https://github.com/gramineproject/gsc/blob/93818d5bc4041e8fabac9c48714e9c6e32c0c89f/config.yaml.template#L17)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

Please add an empty comment line before this `GSC releases ...` paragraph, for readability. Just like above, before the `It is also possible ...` paragraph.
</blockquote></details>

Done

**Review by dimakuv — DISMISSED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gsc/117)* status: all files reviewed, all discussions resolved, not enough approvals from maintainers (2 more required), not enough approvals from different teams (1 more required, approved so far: Intel)

**Review by mkow — CHANGES_REQUESTED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gsc/117)* status: all files reviewed, 5 unresolved discussions, not enough approvals from maintainers (2 more required), not enough approvals from different teams (1 more required, approved so far: Intel) (waiting on @anjalirai-intel)

___
*[`-- commits` line 6 at r3](https://reviewable.io/reviews/gramineproject/gsc/117#-NKrf-Fq1iKoi8D0sPgQ):*
Please just replace with:

> GSC `master` is compatible only with Gramine `master` and GSC may fail if run with Gramine v1.3.1.
___
*[`config.yaml.template` line 18 at r3](https://reviewable.io/reviews/gramineproject/gsc/117#-NKrfSNiASDjmSI1Ju2p:-NKrfSNiASDjmSI1Ju2q:undefined) ([raw file](https://github.com/gramineproject/gsc/blob/8b31ec6708b7c697486ea359b48f0bb59ce2ea21/config.yaml.template#L18)):*
> ```
> #   Image:      "<prebuilt Gramine Docker image>"
> #
> # GSC releases are guaranteed to work with corresponding Gramine releases (and GSC master
> ```





_Suggestion:_
```
`master`
```
___
*[`config.yaml.template` line 19 at r3](https://reviewable.io/reviews/gramineproject/gsc/117#-NKrfU2I7dkKqI7rC8go:-NKrfU2I7dkKqI7rC8gp:undefined) ([raw file](https://github.com/gramineproject/gsc/blob/8b31ec6708b7c697486ea359b48f0bb59ce2ea21/config.yaml.template#L19)):*
> ```
> #
> # GSC releases are guaranteed to work with corresponding Gramine releases (and GSC master
> # branch is guaranteed to work with current Gramine master branch). Therefore, in case of
> ```





_Suggestion:_
```
`master` branch
```
___
*[`config.yaml.template` line 20 at r3](https://reviewable.io/reviews/gramineproject/gsc/117#-NKrfZTl1AiXn_mG4xe7:-NKrfZTl1AiXn_mG4xe8:b-klfzow) ([raw file](https://github.com/gramineproject/gsc/blob/8b31ec6708b7c697486ea359b48f0bb59ce2ea21/config.yaml.template#L20)):*
> ```
> # GSC releases are guaranteed to work with corresponding Gramine releases (and GSC master
> # branch is guaranteed to work with current Gramine master branch). Therefore, in case of
> # issues, specify Gramine Branch that corresponds to your current GSC release (e.g. "v1.3.1"
> ```

Why is this capitalized?



_Code quote:_
```
Branch
```
___
*[`config.yaml.template` line 20 at r3](https://reviewable.io/reviews/gramineproject/gsc/117#-NKrfeCL2eDa0AuOj_CH:-NKrfeCL2eDa0AuOj_CI:bj6lxsg) ([raw file](https://github.com/gramineproject/gsc/blob/8b31ec6708b7c697486ea359b48f0bb59ce2ea21/config.yaml.template#L20)):*
> Therefore, in case of issues, specify Gramine Branch that corresponds to your current GSC release

That's not correct, users should always specify matching releases / branches, even if other configs seem to work.

**Comment by anjalirai-intel** [comment]
*[`-- commits` line 6 at r3](https://reviewable.io/reviews/gramineproject/gsc/117#-NKrf-Fq1iKoi8D0sPgQ):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

Please just replace with:

> GSC `master` is compatible only with Gramine `master` and GSC may fail if run with Gramine v1.3.1.
</blockquote></details>

Done.

**Review by anjalirai-intel — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gsc/117)* status: 0 of 1 files reviewed, 5 unresolved discussions, not enough approvals from maintainers (3 more required), not enough approvals from different teams (2 more required, approved so far: ) (waiting on @dimakuv and @mkow)

___
*[`config.yaml.template` line 18 at r3](https://reviewable.io/reviews/gramineproject/gsc/117#-NKrfSNiASDjmSI1Ju2p:-NKvNVgc6dJpgqLxVhia:b-896fix) ([raw file](https://github.com/gramineproject/gsc/blob/8b31ec6708b7c697486ea359b48f0bb59ce2ea21/config.yaml.template#L18)):*
> ```
> #   Image:      "<prebuilt Gramine Docker image>"
> #
> # GSC releases are guaranteed to work with corresponding Gramine releases (and GSC master
> ```

Done.
___
*[`config.yaml.template` line 19 at r3](https://reviewable.io/reviews/gramineproject/gsc/117#-NKrfU2I7dkKqI7rC8go:-NKvNW8n0heoZJAZE9Lr:b-896fix) ([raw file](https://github.com/gramineproject/gsc/blob/8b31ec6708b7c697486ea359b48f0bb59ce2ea21/config.yaml.template#L19)):*
> ```
> #
> # GSC releases are guaranteed to work with corresponding Gramine releases (and GSC master
> # branch is guaranteed to work with current Gramine master branch). Therefore, in case of
> ```

Done.
___
*[`config.yaml.template` line 20 at r3](https://reviewable.io/reviews/gramineproject/gsc/117#-NKrfZTl1AiXn_mG4xe7:-NKvNXRkArBbnZ8NCdH3:b-896fix) ([raw file](https://github.com/gramineproject/gsc/blob/8b31ec6708b7c697486ea359b48f0bb59ce2ea21/config.yaml.template#L20)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

Why is this capitalized?
</blockquote></details>

Done.
___
*[`config.yaml.template` line 20 at r3](https://reviewable.io/reviews/gramineproject/gsc/117#-NKrfeCL2eDa0AuOj_CH:-NKvNXwM5hs10vnQ4BTI:b-896fix) ([raw file](https://github.com/gramineproject/gsc/blob/8b31ec6708b7c697486ea359b48f0bb59ce2ea21/config.yaml.template#L20)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

> Therefore, in case of issues, specify Gramine Branch that corresponds to your current GSC release

That's not correct, users should always specify matching releases / branches, even if other configs seem to work.
</blockquote></details>

Done.

**Review by dimakuv — CHANGES_REQUESTED** [review]
> Reviewed 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gsc/117)* status: all files reviewed, 5 unresolved discussions, not enough approvals from maintainers (3 more required), not enough approvals from different teams (2 more required, approved so far: ) (waiting on @mkow)

___
*[`config.yaml.template` line 20 at r3](https://reviewable.io/reviews/gramineproject/gsc/117#-NKrfZTl1AiXn_mG4xe7:-NKw0Mpr4k9ijRIYWzRF:b-7h9nyo) ([raw file](https://github.com/gramineproject/gsc/blob/8b31ec6708b7c697486ea359b48f0bb59ce2ea21/config.yaml.template#L20)):*
<details><summary><i>Previously, anjalirai-intel wrote…</i></summary><blockquote>

Done.
</blockquote></details>

I asked it to be capitalized to clearly indicate that `Gramine Branch` are the keywords in this yaml template file.
___
*[`config.yaml.template` line 20 at r3](https://reviewable.io/reviews/gramineproject/gsc/117#-NKrfeCL2eDa0AuOj_CH:-NKw0UYrDbTPCRhHDkp3:b4t836o) ([raw file](https://github.com/gramineproject/gsc/blob/8b31ec6708b7c697486ea359b48f0bb59ce2ea21/config.yaml.template#L20)):*
<details><summary><i>Previously, anjalirai-intel wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Now the text looks redundant and strange... Let's just remove `Even though GSC ...` sentences completely.

**Review by mkow — CHANGES_REQUESTED** [review]
> Reviewed 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gsc/117)* status: all files reviewed, 2 unresolved discussions, not enough approvals from maintainers (3 more required), not enough approvals from different teams (2 more required, approved so far: ) (waiting on @dimakuv)

___
*[`config.yaml.template` line 20 at r3](https://reviewable.io/reviews/gramineproject/gsc/117#-NKrfZTl1AiXn_mG4xe7:-NKx6Fl00Tw81q9_PKLz:b-8pr9r4) ([raw file](https://github.com/gramineproject/gsc/blob/8b31ec6708b7c697486ea359b48f0bb59ce2ea21/config.yaml.template#L20)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

I asked it to be capitalized to clearly indicate that `Gramine Branch` are the keywords in this yaml template file.
</blockquote></details>

Then it should also be in backticks.
___
*[`config.yaml.template` line 20 at r3](https://reviewable.io/reviews/gramineproject/gsc/117#-NKrfeCL2eDa0AuOj_CH:-NKx6P9BBlvETde62Pzw:bon9zbv) ([raw file](https://github.com/gramineproject/gsc/blob/8b31ec6708b7c697486ea359b48f0bb59ce2ea21/config.yaml.template#L20)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

Now the text looks redundant and strange... Let's just remove `Even though GSC ...` sentences completely.
</blockquote></details>

Yeah, now the English/paragraph construction is broken and the whole thing sounds really weird.

**Review by anjalirai-intel — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gsc/117)* status: 0 of 1 files reviewed, 2 unresolved discussions, not enough approvals from maintainers (3 more required), not enough approvals from different teams (2 more required, approved so far: ) (waiting on @dimakuv and @mkow)

___
*[`config.yaml.template` line 20 at r3](https://reviewable.io/reviews/gramineproject/gsc/117#-NKrfeCL2eDa0AuOj_CH:-NKxl12VA2rRdIwnuLJf:b-896fix) ([raw file](https://github.com/gramineproject/gsc/blob/8b31ec6708b7c697486ea359b48f0bb59ce2ea21/config.yaml.template#L20)):*
<details><summary><i>Previously, mkow (Michał Kowalczyk) wrote…</i></summary><blockquote>

Yeah, now the English/paragraph construction is broken and the whole thing sounds really weird.
</blockquote></details>

Done.

**Review by dimakuv — APPROVED** [review]
> Reviewed 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gsc/117)* status: all files reviewed, 2 unresolved discussions, not enough approvals from maintainers (2 more required), not enough approvals from different teams (1 more required, approved so far: Intel) (waiting on @anjalirai-intel and @mkow)
